File: submissions/mar-megint-ti/c9.rb
def f(s)
	t=s.split(\s+)
	&l=3
	&c=0
	&i=0
	&b=false
	$j=0
	while $j < t.length  do
		if t[j].match('^(p|.oo).*') && i<1 then
			if b === true then
				b=false
			else
				l--
				if l<1 
				puts "GAME OVER"
				ret
			end
		end
		if t[j].match('\\d+') then c+=t[j].to_i end
		if c>99 then c-=100 l++ end
		if t[j].match('^(P|B).*') then puts 'WIN' ret end
		if t[j].match("1Up")) then l++ end
		if t[j].match("^M.*")) then b=true end
		i--;
		if t[j].match("Star")) then i=2; end
	end
	puts "???"
end

File: submissions/api-metadata/Columnarize/src/c9.java

public class c9 {

    public static void main(String args[]) {
        f("piranha  fsf Princess   piranha");
        f("1Up koopa goomba piranha Bowser");
        f("Star Mushroom piranha goomba piranha goomba Princess");
        f("piranha piranha piranha piranha");
        f("piranha piranha Star goomba goomba Princess");
        f("Star 1Up 10 40 koopa goomba 20 29 piranha piranha");
        f("koopa 1Up goomba piranha 100 piranha 50 1Up 20 koopa 10 30 piranha 20 40 60 piranha Princess");
    }

    private static void f(String q) {
        int l = 3, i = 0, c = 0;
        boolean d = false, j = false, w = false, b = false;
        String[] h = q.replaceAll("\\s+", " ").trim().split(" ");
        for (String r : h) {
            switch (r) {
                case "Mushroom":
                    j = true;
                    if (i > 0) i--;
                    break;
                case "1Up":
                    l++;
                    if (i > 0) i--;
                    break;
                case "Star":
                    i = 2;
                    break;
                case "goomba":
                case "koopa":
                case "piranha":
                    if (i == 0) {
                        if (j) j = false;
                        else if (l > 1) l--;
                        else d = true;
                    }
                    if (i > 0) i--;
                    break;
                case "Bowser":
                case "Princess":
                    w = true;
                    break;
                default:
                    try {
                        c += Integer.valueOf(r);
                        if (c >= 100) {
                            l++;
                            c -= 100;
                        }
                        if (i > 0) i--;
                    } catch (Exception e) {
                        b = true;
                    }
            }
            if (w) {
                System.out.println("WIN");
                break;
            }
            if (d) {
                System.out.println("GAME OVER");
                break;
            }
            if (b) break;
        }
        if (!d && !w) System.out.println("???");
    }
}